contracts/src/v0.8/dev/keeper2_0/KeeperRegistryBase2_0.sol

```diff
@@ -107,198 +127,275 @@ abstract contract KeeperRegistryBase2_0 is ConfirmedOwner, ExecutionPrevention,
+  event PaymentWithdrawn(address indexed transmitter, uint256 indexed amount, address indexed to, address payee);
   event UpkeepAdminTransferRequested(uint256 indexed id, address indexed from, address indexed to);
   event UpkeepAdminTransferred(uint256 indexed id, address indexed from, address indexed to);
   event UpkeepCanceled(uint256 indexed id, uint64 indexed atBlockHeight);
   event UpkeepCheckDataUpdated(uint256 indexed id, bytes newCheckData);
   event UpkeepGasLimitSet(uint256 indexed id, uint96 gasLimit);
   event UpkeepMigrated(uint256 indexed id, uint256 remainingBalance, address destination);
   event UpkeepPaused(uint256 indexed id);
-  event UpkeepPerformed(
-    uint256 indexed id,
-    bool indexed success,
-    address indexed from,
-    uint96 payment,
-    bytes performData
-  );
+  event UpkeepPerformed(uint256 indexed id, bool indexed success, UpkeepPerformedLogFields upkeepPerformedLogFields);
   event UpkeepReceived(uint256 indexed id, uint256 startingBalance, address importedFrom);
   event UpkeepUnpaused(uint256 indexed id);
   event UpkeepRegistered(uint256 indexed id, uint32 executeGas, address admin);
+  event StaleUpkeepReport(uint256 indexed id);
+  event ReorgedUpkeepReport(uint256 indexed id);
+  event InsufficientFundsUpkeepReport(uint256 indexed id);
+  event CancelledUpkeepReport(uint256 indexed id);
 
   /**
    * @param paymentModel the payment model of default, Arbitrum, or Optimism
-   * @param registryGasOverhead the gas overhead used by registry in performUpkeep
    * @param link address of the LINK Token
-   * @param linkEthFeed address of the LINK/ETH price feed
+   * @param linkNativeFeed address of the LINK/Native price feed
    * @param fastGasFeed address of the Fast Gas price feed
    */
   constructor(
     PaymentModel paymentModel,
-    uint256 registryGasOverhead,
     address link,
-    address linkEthFeed,
+    address linkNativeFeed,
     address fastGasFeed
   ) ConfirmedOwner(msg.sender) {
-    PAYMENT_MODEL = paymentModel;
-    REGISTRY_GAS_OVERHEAD = registryGasOverhead;
-    LINK = LinkTokenInterface(link);
-    LINK_ETH_FEED = AggregatorV3Interface(linkEthFeed);
-    FAST_GAS_FEED = AggregatorV3Interface(fastGasFeed);
+    i_paymentModel = paymentModel;
+    i_link = LinkTokenInterface(link);
+    i_linkNativeFeed = AggregatorV3Interface(linkNativeFeed);
+    i_fastGasFeed = AggregatorV3Interface(fastGasFeed);
+  }
+
+  ////////
+  // GETTERS
+  ////////
+
+  // @dev We want to share these getters on both logic and proxy contract
+  function getPaymentModel() external view returns (PaymentModel) {
+    return i_paymentModel;
+  }
+
+  function getLinkAddress() external view returns (address) {
+    return address(i_link);
+  }
+
+  function getLinkNativeFeedAddress() external view returns (address) {
+    return address(i_linkNativeFeed);
+  }
+
+  function getFastGasFeedAddressl() external view returns (address) {
+    return address(i_fastGasFeed);
```

Comment:
Added these getters in Base since it's better to have them on both logic and proxy